pennylane/templates/subroutines/qsvt.py

```diff
@@ -464,3 +466,290 @@ def _qsp_to_qsvt(angles):
+        update_vals = qml.math.convert_like(update_vals, angles)
+
+        return angles - update_vals
+
+    raise AssertionError(
+        f"Invalid conversion. The conversion between {routine1} --> {routine2} is not defined."
+    )
+
+
+def poly_to_angles(poly, routine, angle_solver="root-finding"):
+    r"""
+    Converts a given polynomial into angles for to be used with specific quantum signal processing (QSP)
+    or quantum singular value transformation (QSVT) routines.
+
+    Args:
+        poly (array-like): coefficients of the polynomial, ordered from lowest to higher degree.
+                           The polynomial must have defined parity and real coefficients.
+                           It also must meet that :math:`|P(x)| \leq 1` for all :math:`x \in [-1, 1]`.
+                           More information about these restriction can be
+                           found in [`arXiv:2105.02859 <https://arxiv.org/abs/2105.02859>`_]
+
+        routine (str): specifies the type of angle transformation required. Must be either: ``"QSP"`` or ``"QSVT"``
+
+        angle_solver (str): optional string which specifies the method used to calculate the angles (must be ``"root-finding"``)
+
+    Returns:
+        (array-like): angles corresponding to the specified transformation routine
+
+    Raises:
+        AssertionError: if poly is not valid in the chosen subroutine
+        AssertionError: if routine or angle_solver specified does not exist
+
+
+    **Example**
+
+    This example applies the polynomial :math:`P(x) = x - \frac{x^3}{2} + \frac{x^5}{3}` to a block-encoding
+    of :math:`x = 0.2`.
+
+    .. code-block::
+
+        poly = [0, 1.0, 0, -1/2, 0, 1/3]
+
+        qsvt_angles = qml.poly_to_angles(poly, "QSVT")
+
+        x = 0.2
+
+        # Encode x in the top left of the matrix
+        block_encoding = qml.RX(2 * np.arccos(x), wires=0)
+        projectors = [qml.PCPhase(angle, dim=1, wires=0) for angle in qsvt_angles]
+
+        @qml.qnode(qml.device("default.qubit"))
+        def circuit_qsvt():
+            qml.QSVT(block_encoding, projectors)
+            return qml.state()
+
+        output = qml.matrix(circuit_qsvt, wire_order=[0])()[0, 0]
+        expected = sum(coef * (x**i) for i, coef in enumerate(poly))
+
+        print("output qsvt: ", output.real)
+        print("P(x) =       ", expected)
+
+    .. code-block:: pycon
+
+        output qsvt:  0.19610666666647059
+        P(x) =        0.19610666666666668
+
+    """
+
+    # Trailing zeros are removed from the array
+    for _ in range(len(poly)):
+        if not np.isclose(poly[-1], 0.0):
+            break
+        poly.pop()
+
+    if len(poly) == 1:
+        raise AssertionError("The polynomial must have at least degree 1.")
+
+    for x in [-1, 0, 1]:
+        if qml.math.abs(qml.math.sum(coeff * x**i for i, coeff in enumerate(poly))) > 1:
+            # It is not a property that we can check globally but checking these three points is useful
```

Comment:
Not sure I understand this.

Reply:
P should mee that |P(x)| ≤ 1
As it is not always easy to detect and the user may not be aware, I am checking that it is true at least  in x = -1, x = 0, x = 1. I updated the comment 👍